extensions/reactive-db2-client/runtime/src/main/java/io/quarkus/reactive/db2/client/runtime/health/ReactiveDB2DataSourceHealthCheck.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.reactive.db2.client.runtime.health;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Readiness;
+
+import io.quarkus.arc.Arc;
+import io.vertx.db2client.DB2Pool;
+
+@Readiness
+@ApplicationScoped
+public class ReactiveDB2DataSourceHealthCheck implements HealthCheck {
+    private DB2Pool db2Pool;
+
+    @PostConstruct
+    protected void init() {
+        db2Pool = Arc.container().instance(DB2Pool.class).get();
```

Comment:
Probably more a question for @mkouba, but what's the reason to not use a regular `@Inject` or an `Instance`?

Reply:
+1 for `@Inject` or `Instance`.

Reply:
This was copied from the other reactive DB health checks, will update this to use `@Inject`